### PR TITLE
Use jwt-keyfile option if available.

### PR DESCRIPTION
**Pull request by lamai93**
This is untested because no image available.

### CODE REVIEW

File: pkg/deployment/resources/pod_creator.go [hunk 1]
```diff
@@ -85,8 +96,17 @@ func createArangodArgs(apiObject metav1.Object, deplSpec api.DeploymentSpec, gro
 		// With authentication
 		options = append(options,
 			optionPair{"--server.authentication", "true"},
-			optionPair{"--server.jwt-secret", "$(" + constants.EnvArangodJWTSecret + ")"},
 		)
+		if versionHasJWTSecretKeyfile(version) {
+			keyPath := filepath.Join(k8sutil.ClusterJWTSecretVolumeName, constants.SecretKeyToken)
```
Comment (ewoutp) [inline pkg/deployment/resources/pod_creator.go]: Don't think using the volume name is correct.
Should be the mount path of that volume (`ClusterJWTSecretVolumeMountDir`)

File: pkg/deployment/resources/pod_creator.go [hunk 2]
```diff
@@ -65,6 +65,17 @@ func versionHasAdvertisedEndpoint(v driver.Version) bool {
 	return v.CompareTo("3.4.0") >= 0
 }
 
+func versionHasJWTSecretKeyfile(v driver.Version) bool {
```
Comment (ewoutp) [inline pkg/deployment/resources/pod_creator.go]: Please add comment

File: pkg/deployment/resources/pod_creator.go (hunk 1 above)
Comment (neunhoef) [reply pkg/deployment/resources/pod_creator.go]: Right.

File: pkg/util/k8sutil/pods.go [hunk 3]
```diff
@@ -447,6 +448,9 @@ func CreateArangodPod(kubecli kubernetes.Interface, developmentMode bool, deploy
 	if rocksdbEncryptionSecretName != "" {
 		c.VolumeMounts = append(c.VolumeMounts, rocksdbEncryptionVolumeMounts()...)
 	}
+	if ClusterJWTSecretVolumeName != "" {
```
Comment (neunhoef) [inline pkg/util/k8sutil/pods.go]: I think this will always be true.

Comment (neunhoef) [reply pkg/util/k8sutil/pods.go]: Should be `clusterJWTSecretName`

File: pkg/deployment/resources/pod_creator.go (hunk 2 above)
Comment (neunhoef) [reply pkg/deployment/resources/pod_creator.go]: Done.

File: pkg/util/k8sutil/pods.go (hunk 3 above)
Comment (lamai93) [reply pkg/util/k8sutil/pods.go]: Yes

**Pull request merged**